### ISSUE TITLE
Minimal working example for TabNetRegressor/Classifier

### ISSUE THREAD

**Issue by SurtaiHan** [dreamquark-ai/tabnet#519]
Hi, I'm trying to make a minimal working example to understand how to use TabNet.

Currently, with the features as the identity matrix of size 3, and the target values as a vector [3,4,5], the final fitted model seems to predict negative values.

**If the current behavior is a bug, please provide the steps to reproduce.**
The full code to reproduce the issue is shown below, along with the current output.

**Expected behavior**
I expect it to correctly produce [3,4,5] on the given input because it seems like this should be learnable even with a linear regression. Although I'm not sure if this expectation is unrealistic for DNNs.

**Screenshots**
![image](https://github.com/dreamquark-ai/tabnet/assets/5789945/40aea2cc-e87d-4e3b-897e-84a516803bc9)

![image](https://github.com/dreamquark-ai/tabnet/assets/5789945/f0214f5a-35af-4bb1-ad37-bbfc5d850e3e)

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

**Comment by SurtaiHan** [comment]
I'm also a bit confused why loss=0 makes sense alongside a non-zero mse.

**Comment by SurtaiHan** [comment]
Seems like when batch_size is greater than the number of datapoints, loss is 0. When I set batch size to 2, it works for the classifier case.

**Comment by Optimox** [comment]
During training, drop_last=True for the dataloader, so if your entire dataset is smaller than your batch size then the model is never trained at all. loss=0 is because it is instanciated with the value 0.

**Comment by SurtaiHan** [comment]
that worked, thanks!
